add 3.14 support for py3-pip

kranurag7 opened the pull request:
Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
add 3.14 support for py3-pip

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
add 3.14 support to setuptools

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
chore: remove py3-foo type provides

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

Taffer reviewed (dismissed):
Assuming this just makes the 3.14 versions of `pip` and `setuptools` available, I think this is right. How's that for an endorsement? 🤪

kranurag7 committed:
chore: remove provider-priority from config

this is not required given we're not longer using provides so it's safe
to remove provider-priority as well.

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>

kranurag7 committed:
address review comments

uses python-3.14-base at build time
adds test/tw/pip-check during testing
removes value for python-versions given we don't use range.value anywhere

Signed-off-by: kranurag7 <81210977+kranurag7@users.noreply.github.com>
